e2e/nomostest/prometheus_metrics.go

```diff
@@ -344,56 +344,54 @@ func metricResourceFightsHasValueAtLeast(nt *NT, syncLabels prometheusmodel.Labe
 		labels := prometheusmodel.LabelSet{
 			prometheusmodel.LabelName(ocmetrics.KeyComponent.Name()): prometheusmodel.LabelValue(ocmetrics.OtelCollectorName),
 		}.Merge(syncLabels)
-		// ResourceFightsView counts the total number of ResourceFights, so we don't need to aggregate
-		query := fmt.Sprintf("%s%s", metricName, labels)
 		if value == 0 {
-			// Tolerate missing metrics when expecting a zero value.
-			// Don't allow any value other than zero.
-			return metricExistsWithValueOrDoesNotExist(ctx, nt, v1api, query, float64(value))
+			query := fmt.Sprintf("rate(%s%s[1m])", metricName, labels)
+			return metricExistsWithValue(ctx, nt, v1api, query, 0)
 		}
+		query := fmt.Sprintf("%s%s", metricName, labels)
```

Comment:
why are we using rate for `metricExistsWithValue` but not `metricExistsWithValueAtLeast`?

Reply:
When it falls in value == 0 check,  the rsync summary passed in contains no conflict / fight / internal error. In this case we want the recording rate of this metric to be strictly 0 to indicate that no conflict / fight / internal error happening.

When the given value is >0, the aforementioned error has happened and is gonna increment as retry happens, so we verify that the actual counter is at least the number of times that we think it's happening. Rate in this case would just report a number like 0.87/s, which doesn't mean much when verifying.

Reply:
I think if we set the value=0 when initializing the metrics and then modify the query to only match value=1 before the rate, then we can unify these.

Reply:
I'm confused, we are not setting value=0. We could do match value=1 then check rate=0, but then we still need to check for valueAtLeast when there's fight happening.

Reply:
What I'm saying is that we COULD set value=0 when we initialize the metric. Then it would be possible to ignore those records when querying the rate, by only querying for value=1.

Reply:
This part has been moved to https://github.com/GoogleContainerTools/kpt-config-sync/pull/1237, will be making change accordingly.